dss/api/bundles/__init__.py

```diff
@@ -132,6 +136,50 @@ def get(
         response.headers['Link'] = link
         return response
 
+@dss_handler
+def enumerate(replica: str, prefix: str, token: str = None,
+              per_page: int = PerPageBounds.per_page_max,
+              search_after: typing.Optional[str] = None):
+
+    storage_handler = Config.get_blobstore_handle(Replica[replica])
+    key_prefix = prefix if prefix else 'bundles/'  # might need to add filtering over here (no files/blobs allowed)
+    """Does this even need to have prefix? the only option is bundles, only in the checkout bucket does the
+        Bundles/ prefix actually have files, and thats after a --presigned url, otherwise contents after bundles/ are not
+        exposed........"""
+    kwargs = dict(bucket=Replica[replica].bucket, prefix=key_prefix, k_page_max=per_page)
+    if search_after:
+        kwargs['start_after_key'] = search_after
+    if token:
+        kwargs['token'] = token
+    prefix_iterator = storage_handler.list_v2(**kwargs)
```

Comment:
`list_v2` accepts an argument to limit the number of keys returned: `k_page_max`. This can be used to avoid the `StopIteration` logic below.

Reply:
This.  ^